lib/vwo/core/variation_decider.rb

```diff
@@ -53,9 +54,54 @@ def initialize(settings_file, user_storage_service = nil)
       # @return[String,String]                       ({variation_id, variation_name}|Nil): Tuple of
       #                                              variation_id and variation_name if variation allotted, else nil
 
-      def get_variation(user_id, campaign, campaign_key, custom_variables = {})
+      def get_variation(user_id, campaign, campaign_key, custom_variables = {}, variation_targeting_variables = {})
         campaign_key ||= campaign['key']
 
+        if campaign['isForcedVariationEnabled']
+          variation = evaluate_Whitelisting(
+            user_id,
+            campaign,
+            campaign_key,
+            variation_targeting_variables
+          )
+          if variation
+            status = StatusEnum::PASSED
+            variation_string = variation['name']
+          else
+            status = StatusEnum::FAILED
+            variation_string = ''
+          end
+
+          @logger.log(
+            LogLevelEnum::INFO,
+            format(
+              LogMessageEnum::InfoMessages::SEGMENTATION_STATUS,
+              file: FILE,
+              campaign_key: campaign_key,
+              user_id: user_id,
+              status: status,
+              custom_variables: variation_targeting_variables,
+              variation_name: variation_string,
+              segmentation_type: SegmentationTypeEnum::WHITELISTING,
+              api_name: ApiMethods::GET_FEATURE_VARIABLE_VALUE,
+            )
+          )
+          if variation
```

Comment:
return variation if variation

Reply:
need to check variation['name'] too. changed accordingly